### PR TITLE
DEP: Deprecate the keepdims argument to accumulate

### PR DESCRIPTION
**Pull request by eric-wieser**
As discussed in #8187

### CODE REVIEW

**Comment by charris** [comment]
OK, looks good.  Two more things, add an entry to the `doc/release/1.12.0-notes.rst` mentioning the deprecation and then put a test in `numpy/cores/tests/test_deprecations.py`. It is also helpful if the numpy version and date are in a comment before the deprecation so that it can be searched for and checked when the time comes. Something like
```
/* 2016-10-21, 1.12 */
```

**Comment by charris** [comment]
At this point you must feel like you are getting nickle and dimed to death ;)  Thanks for your patience. Changing arguments/behavior is an ambitious thing...

File: numpy/core/src/umath/ufunc_object.c [hunk 1]
```diff
@@ -3891,18 +3894,29 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         }
     }
     else if (operation == UFUNC_ACCUMULATE) {
-        if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|OO&O&i", kwlist1,
+        PyObject *bad_keepdimarg;
```
Comment (charris) [inline numpy/core/src/umath/ufunc_object.c]: Need to initialize to `NULL`.

**Comment by eric-wieser** [comment]
What exactly is this comment trying to acheive? Doesn't a `git blame` contain that information?

**Comment by eric-wieser** [comment]
Updated with tests and a NULL initialization

File: numpy/core/tests/test_deprecations.py [hunk 2]
```diff
@@ -631,6 +631,13 @@ def test_all_dtypes(self):
             self.assert_deprecated(np.dtype, exceptions=(TypeError,),
                                    args=(dt,))
 
+class TestAccumulateKeepDims(_DeprecationTestCase):
+    """
+    Deprecate the keepdims argument to np.ufunc.accumulate, which was never used or documented
+    """
+    def test_keepdims(self):
```
Comment (charris) [inline numpy/core/tests/test_deprecations.py]: Want this:
```
        with warnings.catch_warnings():
            warnings.filterwarnings('always', '', FutureWarning)
            assert_warns(FutureWarning, np.add.accumulate, [1], keepdims=True)
```

**Comment by charris** [comment]
Thanks @eric-wieser .

**Pull request merged**